core/src/main/java/org/apache/iceberg/rest/ErrorHandlers.java

```diff
@@ -0,0 +1,124 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.util.function.Consumer;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.BadRequestException;
+import org.apache.iceberg.exceptions.ForbiddenException;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.NotAuthorizedException;
+import org.apache.iceberg.exceptions.RESTException;
+import org.apache.iceberg.exceptions.ServiceFailureException;
+import org.apache.iceberg.rest.responses.ErrorResponse;
+
+/**
+ * A set of consumers to handle errors for requests for table entities or for namespace entities,
+ * to throw the correct exception.
+ */
+public class ErrorHandlers {
+
+  private ErrorHandlers() {
+  }
+
+  public static Consumer<ErrorResponse> namespaceErrorHandler() {
+    return baseNamespaceErrorHandler().andThen(defaultErrorHandler());
+  }
+
+  public static Consumer<ErrorResponse> tableErrorHandler() {
+    return baseTableErrorHandler().andThen(defaultErrorHandler());
+  }
+
+  /**
+   * Table level error handlers.
+   * Should be chained wih the {@link #defaultErrorHandler}, which takes care of common cases.
+   */
+  private static Consumer<ErrorResponse> baseTableErrorHandler() {
+    return error -> {
+
+      switch (error.code()) {
+        // HttpStatus.SC_NOT_FOUND
+        case 404:
+          if (NoSuchNamespaceException.class.getSimpleName().equals(error.type())) {
+            throw new NoSuchNamespaceException("Resource not found: %s", error);
+          } else {
+            throw new NoSuchTableException("Resource not found: %s", error);
+          }
+        // HttpStatus.SC_CONFLICT
+        case 409:
+          throw new AlreadyExistsException("The table already exists: %s", error);
+      }
+    };
+  }
+
+  /**
+   * Request error handlers specifically for CRUD ops on namespaces.
+   * Should be chained wih the {@linkplain ErrorHandlers#defaultErrorHandler}, which takes care of common cases.
+   */
+  private static Consumer<ErrorResponse> baseNamespaceErrorHandler() {
+    return error -> {
+      switch (error.code()) {
+        // HttpStatus.SC_NOT_FOUND
+        case 404:
+          throw new NoSuchNamespaceException("Namespace not found: %s", error);
+        // HttpStatus.SC_CONFLICT
+        case 409:
+          throw new AlreadyExistsException("Namespace already exists: %s", error);
+        // HttpStatus.SC_UNPROCESSABLE_ENTITY
+        case 422:
+          throw new RESTException("Unable to process request due to bad property updates");
+      }
+    };
+  }
+
+  /**
+   * Request error handler that handles the common cases that are included with all responses,
+   * such as 400, 401, 403, 500, etc.
+   */
+  public static Consumer<ErrorResponse> defaultErrorHandler() {
+    return errorResponse -> {
```

Comment:
The handlers above use `error` instead of `errorResponse`.

Reply:
Updated!